src/web/lib/components/ThemeColorsEditor/index.js

```diff
@@ -2,28 +2,65 @@ import React from "react";
 import classnames from "classnames";
 import { SketchPicker } from "react-color";
 import onClickOutside from "react-onclickoutside";
-import { colorLabels, colorsWithAlpha } from "../../../../lib/constants";
+import { colorLabels, colorsWithAlpha, ESC } from "../../../../lib/constants";
 import { colorToCSS } from "../../../../lib/themes";
 import Metrics from "../../../../lib/metrics";
 
 import "./index.scss";
 
 class ThemeColorsEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: false,
+      colorName: null
+    };
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+  }
+
+  handleClick(name) {
+    const { setSelectedColor } = this.props;
+    setSelectedColor({ name });
+    if (this.state.selected && this.state.colorName !== name) {
```

Comment:
Here, we should be able to check if `name === this.props.selectedColor` to decide whether to `setSelectedColor({ name: null })` which should clear the current selected color